classes/Mock.php

```diff
@@ -188,15 +196,84 @@ public function define()
                 
             use malkusch\phpmock\MockCallHelper;
 
-            function $this->name()
+            function $this->name({$this->getParametersList()})
             {
+                \$arguments = {$this->getArgumentsList()};
                 return MockCallHelper::call(
                     '$this->name',
                     '$canonicalFunctionName',
-                    func_get_args()
+                    \$arguments
                 );
             }";
-        
+
         eval($definition);
     }
+
+    /**
+     * Get a list of parameters for the function-definition
+     *
+     * @return string
+     */
+    private function getParametersList()
+    {
+        $functionReflection = new ReflectionFunction('\\' . $this->name);
+        $argsReflection = $functionReflection->getParameters();
+        $arguments = array();
+
+        foreach ($argsReflection as $arg) {
+            if ($arg->name == '...') {
+                // If '...' is set as name of a parameter this is a variadic
+                // C-implementation before PHP5.5 - There is no way of knowing
+                // how many parameters there might be given, so lets simply
+                // use the func_get_args for getting the params. As these
+                // "variadic" functions do not use pas-by-reference it doesn't
+                // matter.
+                return '';
+            }
+            $argument = '';
+            if (true === $arg->isPassedByReference()) {
+                $argument .= '&';
+            }
+            $argument .= '$' . $arg->name;
+
+            if ($arg->isOptional()) {
+                $argument .= ' = \'optionalParameter\'';
+            }
+            $arguments[ ] = $argument;
+        }
+
+        return implode(', ', $arguments);
+    }
+
+    /**
+     * Get a string representation of the params array
+     *
+     * @return string
+     */
+    private function getArgumentsList()
+    {
+        $functionReflection = new ReflectionFunction('\\' . $this->name);
+        $argsReflection = $functionReflection->getParameters();
+        $arguments = array();
+
+        foreach ($argsReflection as $arg) {
+            if ($arg->name == '...') {
+                // If '...' is set as name of a parameter this is a variadic
+                // C-implementation before PHP5.5 - There is no way of knowing
```

Comment:
I'm a bit outdated in PHP. Give me a hint what you mean by "this is a variadic C-implementation before PHP5.5". I understand that [`...`](http://php.net/manual/en/functions.arguments.php#functions.variable-arg-list.new) came with PHP-5.6. What is this variadic C-implementation before PHP-5.5?